extension-requests/script.js

```diff
@@ -913,76 +988,155 @@ async function createExtensionCard(data) {
-        <div class="log-div"> 
-        <img class="log-img" src="/images/edit-icon.png"></img>
-          <p class="reason-text">${
-            log?.meta?.userId === currentUserDetails.id
-              ? 'You'
-              : log?.meta?.name || 'Super User'
-          } changed the title from ${log.body.oldTitle} to ${log.body.newTitle}.
-          </p>
-          </div>`);
+        sentence = checkIfPreviouslyRendered(id, 'TITLE', log, parentClassName);
+        sentence && arraySentence.push(sentence);
       }
     });
 
   return arraySentence.reverse().join('');
 }
+
+function checkIfPreviouslyRendered(id, logType, log, parentClassName) {
+  const alreadyRenderdLogs = renderLogRecord[id] || [];
+  let sentence = '';
+  let text = '';
+  switch (logType) {
+    case 'REVIEW': {
+      const updationTime = dateDiff(
+        Date.now(),
+        secondsToMilliSeconds(log?.timestamp?._seconds),
+      );
+      text = `${
+        log?.meta?.userId === currentUserDetails.id
+          ? 'You'
+          : log?.meta?.name || 'Super User'
+      } ${log.body.status} this request ${updationTime} ago.`;
+
+      sentence = `
+      <div class="log-div ${parentClassName}">
+        <img class="log-img" src="/images/${
+          log?.body?.status === 'APPROVED' ? 'approved.png' : 'denied.png'
+        }">
+        </img>
+        <p class="reason-text">${text}</p>
+      </div>
+      `;
+      break;
+    }
+    case 'ETA': {
+      text = `${
+        log?.meta?.userId === currentUserDetails.id
+          ? 'You'
+          : log?.meta?.name || 'Super User'
+      } changed the ETA from ${fullDateString(
+        secondsToMilliSeconds(log.body.oldEndsOn),
+      )} to ${fullDateString(secondsToMilliSeconds(log.body.newEndsOn))}.`;
+      sentence = `
+      <div class="log-div ${parentClassName}">
+        <img class="log-img" src="/images/edit-icon.png"></img>
+        <p class="reason-text">
+        ${text}
+        </p>
+      </div>
+      `;
+      break;
+    }
+    case 'REASON': {
+      text = `${
+        log?.meta?.userId === currentUserDetails.id
+          ? 'You'
+          : log?.meta?.name || 'Super User'
+      } changed the reason from ${log.body.oldReason} to ${
+        log.body.newReason
+      }.`;
+      sentence = `
+      <div class="log-div ${parentClassName}"> 
+        <img class="log-img" src="/images/edit-icon.png"></img>
+        <p class="reason-text">${text}</p>
+      </div>
+      `;
+      break;
+    }
+
+    case 'TITLE': {
+      text = `${
+        log?.meta?.userId === currentUserDetails.id
+          ? 'You'
+          : log?.meta?.name || 'Super User'
+      } changed the title from ${log.body.oldTitle} to ${log.body.newTitle}.`;
+      sentence = `
+      <div class="log-div ${parentClassName}"> 
+        <img class="log-img" src="/images/edit-icon.png"></img>
+        <p class="reason-text">
+        ${text}
+        </p>
+      </div>`;
+    }
+  }
+  if (alreadyRenderdLogs.includes(text)) {
```

Comment:
what if  an empty string if the text variable is already present in the alreadyRenderdLogs array. This means that if the same log message is rendered twice, the second time it is rendered, it will be empty.

can you please let me your thought about below code 

if (alreadyRenderdLogs.includes(text)) {
 return alreadyRenderdLogs[alreadyRenderdLogs.indexOf(text)];
}

Reply:
The text will only be inserted into alreadyRenderdLogs if it has been rendered. In the case of an empty string, it will not be rendered into the UI and ultimately won't be pushed into alreadyRenderdLogs.